cmake/FindR.cmake

```diff
@@ -38,50 +38,74 @@ endif()
 set(TEMP_CMAKE_FIND_APPBUNDLE ${CMAKE_FIND_APPBUNDLE})
 set(CMAKE_FIND_APPBUNDLE "NEVER")
 find_program(R_COMMAND R DOC "R executable.")
+message(STATUS "R_COMMAND found at: ${R_COMMAND}")
 find_program(R_SCRIPT_COMMAND Rscript DOC "Rscript executable.")
+message(STATUS "R_SCRIPT_COMMAND found at: ${R_SCRIPT_COMMAND}")
 set(CMAKE_FIND_APPBUNDLE ${TEMP_CMAKE_FIND_APPBUNDLE})
 
 if(R_COMMAND)
   # temporarily append ".dll" to the cmake find_library suffixes
   set(OLD_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
   set(CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES} ".dll")
 
-  execute_process(WORKING_DIRECTORY .
-                  COMMAND ${R_COMMAND} RHOME
-                  OUTPUT_VARIABLE R_ROOT_DIR
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+  if(CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
+    message(STATUS "Configuring for Emscripten...")
 
-  set(R_HOME ${R_ROOT_DIR} CACHE PATH "R home directory obtained from R RHOME")
-  
-  execute_process(WORKING_DIRECTORY .
-                  COMMAND ${R_COMMAND} CMD config --ldflags
-                  OUTPUT_VARIABLE R_LDFLAGS
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-  set(R_LDFLAGS ${R_LDFLAGS} CACHE PATH "R CMD config --ldflags")
+    set(R_HOME "${CMAKE_PREFIX_PATH}/lib/R" CACHE PATH "R home directory for Emscripten")
 
-  find_path(R_INCLUDE_DIR R.h
-            HINTS ${R_ROOT_DIR} ${R_ROOT_DIR}/bin/${R_LIB_ARCH}
-            PATHS /usr/local/lib /usr/local/lib64 /usr/share
-            PATH_SUFFIXES include R/include
-            DOC "Path to file R.h")
+    set(R_INCLUDE_DIR "${R_HOME}/include" CACHE PATH "Path to R include directory")
 
-  find_library(R_LIBRARY_BASE R
-            HINTS ${R_ROOT_DIR}/lib ${R_ROOT_DIR}/bin/${R_LIB_ARCH}
-            DOC "R library (example libR.a, libR.dylib, etc.).")
+    set(R_LDFLAGS "-Wl,--export-dynamic -fopenmp -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -L${R_HOME}/lib -lR -lpcre2-8 -ldeflate -llzma -lbz2 -lz -ltirpc -lrt -ldl -lm -licuuc -licui18n" CACHE STRING "Linker flags for R libraries in Emscripten")
```

Comment:
These ld flags are questionable... :thinking: 
When checking `r-base`, we should have something similar to this

> ```-sWASM_BIGINT -sMAIN_MODULE=1 -sSTACK_SIZE=5MB -sALLOW_MEMORY_GROWTH=1 -sEXPORTED_RUNTIME_METHODS=callMain,FS,ENV,getEnvStrings,TTY -sFORCE_FILESYSTEM=1 -sINVOKE_RUN=0 -L$PREFIX/lib -lRblas -lFortranRuntime -lpcre2-8 -llzma -lbz2 -lz -lrt -ldl -lm -liconv```

but we probably don't need all of them.

Reply:
Yes for sure, I need to confirm these 
These are just that once that I got through `R_COMMAND` using something like this 
```
    execute_process(WORKING_DIRECTORY .
                    COMMAND ${R_COMMAND} CMD config --ldflags
                    OUTPUT_VARIABLE R_LDFLAGS
                    OUTPUT_STRIP_TRAILING_WHITESPACE)
```
I just pasted it for more reasbility.